insights/parsers/cryptsetup_luksDump.py

```diff
@@ -0,0 +1,175 @@
+"""
+LUKS_Dump - command ``cryptsetup luksDump``
+============================================================
+This class provides parsing for the output of cryptsetup luksDump
+<device_name>. Outputs from LUKS1 and LUKS2 are supported.
+"""
+
+from insights import parser, Parser
+from insights.specs import Specs
+import string
+
+from insights.parsr import Literal, AnyChar, Char, Opt, String, WS, HangingString, WithIndent, Many
+
+
+def convert_type(section):
+    section[1]["type"] = section[0][1]
+    return [section[0][0], section[1]]
+
+
+def convert_status(section):
+    section[2]["status"] = section[1]
+    return [section[0], section[2]]
+
+
+value_chars = set(string.printable) - set("\n\r")
+
+FirstLine = Literal("LUKS header information", value="header") << AnyChar.until(Char("\n")) + Opt(Many(Char("\n")))
```

Comment:
Could you please review the httpd_conf parser, and add all of the parsr logic in a DocParser class like the httpd_conf parser. That way all of the logic is wrapped in a class so it's better organized and has an exceptions catch.